### ISSUE TITLE
Query on LSI returns incorrect results

### ISSUE THREAD

**Issue by lukiano** [mhart/dynalite#61]
I create a table with columns A, B, C where A is hash key, B is range key, and there's also a local index with A as hash key and C as range key.
Then I insert some rows having all 3 columns (and a few more)
Doing a query on the local index, using A = 'a' and C <= 'c' is returning incorrect results (empty results in my case).
If I do C = 'c', then it returns correct results.
Also, doing the inequality comparison in Amazon DynamoDB Local returns the expected results as well.
If it's useful, both A and B are strings, and C is a number.

**Comment by mhart** [comment]
Hmmm, that doesn't sound good!

So you're using [Query](http://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_Query.html) – are you using `KeyConditionExpression` or `KeyConditions`? Any chance you can post the full Query JSON request you're using?

**Comment by lukiano** [comment]
```json
{
   "TableName":"items",
   "IndexName":"SequenceInserted",
   "Limit":null,
   "ConsistentRead":true,
   "ScanIndexForward":true,
   "ExclusiveStartKey":null,
   "ReturnConsumedCapacity":"TOTAL",
   "FilterExpression":null,
   "KeyConditionExpression":"CollectionId = :id AND InsertedAtSequenceId <= :seq",
   "ExpressionAttributeValues":{
      ":id":{
         "S":"7d5175f7-c75b-4bb0-83de-a55bb0f048e9"
      },
      ":seq":{
         "N":"1"
      }
   }
}
```

**Comment by lukiano** [comment]
CollectionId is the hash key for both the table and the local index.  InsertedAtSequenceId is the range key for the local index

**Comment by mhart** [comment]
Cool, thanks! 👍  I'll check this out tomorrow

**Comment by mhart** [comment]
@lukiano thanks for reporting this! Was able to reproduce easily – not sure why I didn't have a test for this!

Anyway, I do now – and it's fixed in v1.0.3

**Comment by lukiano** [comment]
Yay! You're quick!
